src/storage/mapping/FixedContentTypeMapper.ts

```diff
@@ -17,67 +17,88 @@ import { getAbsolutePath, getRelativePath, validateRelativePath } from './Mapper
 export class FixedContentTypeMapper implements FileIdentifierMapper {
   protected readonly logger = getLoggerFor(this);
 
-  private readonly baseRequestURI: string;
-  private readonly rootFilepath: string;
-  private readonly contentType: string;
+  protected readonly baseRequestURI: string;
+  protected readonly rootFilepath: string;
+  protected readonly contentType: string;
 
   public constructor(base: string, rootFilepath: string, contentType: string) {
     this.baseRequestURI = trimTrailingSlashes(base);
     this.rootFilepath = trimTrailingSlashes(normalizeFilePath(rootFilepath));
     this.contentType = contentType;
   }
 
+  /**
+   * Maps the given resource identifier / URL to a file path.
+   * Determines the content-type if no content-type was provided.
+   * For containers the content-type input gets ignored.
+   * @param identifier - The input identifier.
+   * @param contentType - The (optional) content-type of the resource.
+   *
+   * @returns A ResourceLink with all the necessary metadata.
+   */
   public async mapUrlToFilePath(identifier: ResourceIdentifier, contentType?: string): Promise<ResourceLink> {
     const path = getRelativePath(this.baseRequestURI, identifier);
     validateRelativePath(path, identifier);
 
     const filePath = getAbsolutePath(this.rootFilepath, path);
+    return isContainerIdentifier(identifier) ?
+      this.mapUrlToContainerPath(identifier, filePath) :
+      this.mapUrlToDocumentPath(identifier, filePath, contentType);
```

Comment:
I would expect a function called `mapUrlToPath` to take as input a url and output a path. These take as input a url and a path, and output a ResourceLink. But `mapUrlAndFilePathToContainerResourceLink` is not a great name so might need something shorter.

Reply:
What is the reason of existence for `ResourceLink`? If I'm not mistaken, it will always pass one of its argument straight through.

Reply:
I think it was just made as a helper object that contains all the relevant information for a file-based resource. The mapper here could also be changed to only return the new information and then that could be coupled in the `FileDataAccessor` instead (since it's useful there to have an object containing all this information). I'm assuming the decision as it is now was made because we needed a new interface anyway to return both the content-type and the identifier/filepath.